src/list-in.cpp

```diff
@@ -136,66 +136,75 @@ struct Outlet
     db::asset::ExtAttrValue label;
     db::asset::ExtAttrValue type;
     db::asset::ExtAttrValue group;
+    db::asset::ExtAttrValue name;
+    db::asset::ExtAttrValue switchable;
 };
 
-static std::string getOutletNumber(const std::string& extAttributeName)
+// parse ext. attribute as "outlet.<number>.<property-name>
+// returns <outlet number, property name> (empty if inconsistent)
+
+static std::pair<std::string, std::string> getOutletNumberAndProperty(const std::string& extAttributeName)
 {
-    auto        dot1    = extAttributeName.find_first_of(".");
-    std::string oNumber = extAttributeName.substr(dot1 + 1);
-    auto        dot2    = oNumber.find_first_of(".");
-    oNumber             = oNumber.substr(0, dot2);
-    return oNumber;
+    std::pair<std::string, std::string> result; // empty
+
+    if (extAttributeName.find("outlet.") != 0)
+        return result; // empty
+    auto dot1 = extAttributeName.find_first_of(".");
+    if (dot1 == std::string::npos)
+        return result; // empty
+    std::string aux = extAttributeName.substr(dot1 + 1);
+    auto dot2 = aux.find_first_of(".");
+    if (dot2 == std::string::npos)
+        return result; // empty
+
+    result.first = aux.substr(0, dot2); // number
+    result.second = aux.substr(dot2 + 1); // property name
+    return result;
 }
 
-static std::map<std::string, Outlet> collectOutlets(const db::asset::Attributes& ext)
+// returns a ref. to existing Outlet entry in outlets
+
+static Outlet* findOrCreateOutlet(const std::string& key, std::map<std::string, Outlet>& outlets)
```

Comment:
This function is useless: operator[] does already this see https://en.cppreference.com/w/cpp/container/map/operator_at

Running this code is a proof of it:
`
#include <iostream>
#include <string>
#include <map>
 
int main()
{
    std::map<int, std::string>  myMap;
    myMap[0] = "data0";
    std::cout << "Map at 0: '" << myMap[0] <<"'"<< std::endl;
    
    auto & element0 = myMap[0];
    element0 = "new data";
    std::cout << "Map at 0: '" << element0 <<"'"<< std::endl;
    
    auto & element1 = myMap[1];
    std::cout << "Map at 1: '" << element1 <<"'"<< std::endl;
    element1 = "data1";
    std::cout << "Map at 1: '" << element1 <<"'"<< std::endl;
}

`

Output:

> Map at 0: 'data0'
> Map at 0: 'new data'
> Map at 1: ''
> Map at 1: 'data1'